### PR TITLE
Fix a notification message when a guest user tries to create a new re…

### LINKED ISSUE #299: Wrong notification message when a guest user tries to create a repository
See https://github.com/line/centraldogma/blob/master/server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/AbstractRoleCheckingDecorator.java#L62

### PR DESCRIPTION
**Pull request by hyangtack**
…pository

Motivation:
A guest user can be confused because a notification when he or she tries to create a new repository is wierd like below.
`You must be GUEST of project '...'`

Modifications:
- Fix the notification message.
- Refactor decorators using a custom decorator annotation.
  - Replace `@Decorator(...)` with `@RequiresRole`, `@RequiresReadPermission`, `@RequiresWritePermission` and `@RequiresAdministrator`.

Result:
- Closes #299

### CODE REVIEW

File: server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java [hunk 1]
```diff
@@ -372,6 +368,12 @@ private static Object handleWatchFailure(Throwable thrown) {
         return converter.apply(Iterables.getOnlyElement(collection));
     }
 
+    /**
+     * GET /projects/{projectName}/repos/{repoName}/merge?
+     * revision={revision}&amp;path={path}&amp;optional_path={optional_path}
+     *
+     * <p>Returns a merged entry of files which are specified in the query string.
+     */
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java]: Thanks~!

File: server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java [hunk 2]
```diff
@@ -213,14 +212,13 @@ private static String normalizePath(String path) {
      * {@link HttpStatus#NOT_MODIFIED} otherwise.
      */
     @Get("regex:/projects/(?<projectName>[^/]+)/repos/(?<repoName>[^/]+)/contents(?<path>(|/.*))$")
-    @Decorator(HasReadPermission.class)
     public CompletableFuture<?> getFiles(@Param("path") String path,
                                          @Param("revision") @Default("-1") String revision,
                                          Repository repository,
                                          @RequestConverter(WatchRequestConverter.class)
-                                         Optional<WatchRequest> watchRequest,
+                                                 Optional<WatchRequest> watchRequest,
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java]: Could you change this back?

File: server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/RequiresRoleDecorator.java [hunk 3]
```diff
@@ -56,13 +73,14 @@ public final HttpResponse serve(Service<HttpRequest, HttpResponse> delegate, Ser
                 if (cause != null) {
                     return handleException(cause);
                 }
-                if (!isAccessAllowed(ctx, req, user, role)) {
+                if (!user.isAdmin() && !accessibleRoles.contains(role)) {
                     return HttpApiUtil.throwResponse(
                             HttpStatus.FORBIDDEN,
-                            "You must be %s of project '%s'.", role, projectName);
+                            "You must be granted one of the following roles in project '%s': %s",
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/RequiresRoleDecorator.java]: in the project '%s'?
How about `You must have one of the following roles in the project '%s' to access: %s`?

Comment (trustin) [reply server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/RequiresRoleDecorator.java]: How about: `You must have one of the following roles to access the project '%s': %s`

Comment (minwoox) [reply server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/RequiresRoleDecorator.java]: Yeah, it's better :)

**Comment by hyangtack** [comment]
Thanks, reviewers!

**Pull request merged**